graylog2-server/src/main/java/org/graylog2/rest/resources/entities/titles/EntityTitleService.java

```diff
@@ -0,0 +1,137 @@
+                .stream()
+                .map(entry -> getTitlesForEntitiesFromSingleCollection(subject, entry.getKey(), entry.getValue()))
+                .reduce(EntitiesTitleResponse::merge);
+
+        return entitiesTitleResponse.orElse(EMPTY_RESPONSE);
+    }
+
+    private EntitiesTitleResponse getTitlesForEntitiesFromSingleCollection(final Subject subject,
+                                                                           final String collection,
+                                                                           final List<EntityIdentifier> entities) {
+        final Optional<DbEntityCatalogEntry> dbEntityCatalogEntry = this.entitiesCatalog.getByCollectionName(collection);
+        if (dbEntityCatalogEntry.isEmpty() || entities.isEmpty()) {
+            return EMPTY_RESPONSE;
+        }
+
+        final String titleField = dbEntityCatalogEntry.get().titleField();
+        if (titleField.equals(NO_TITLE)) {
+            return new EntitiesTitleResponse(
+                    entities.stream()
+                            .map(e -> new EntityTitleResponse(e.id(), e.type(), NO_TITLE))
+                            .collect(Collectors.toSet()),
+                    Set.of()
+            );
+        }
+
+        final MongoCollection<Document> mongoCollection = mongoConnection.getMongoDatabase().getCollection(collection);
+
+        Bson bsonFilter = Filters.or(
+                entities.stream()
+                        .map(e -> Filters.eq("_id", new ObjectId(e.id())))
+                        .collect(Collectors.toList())
+        );
+
+        final FindIterable<Document> documents = mongoCollection
+                .find(bsonFilter)
+                .projection(Projections.include(titleField));
+
+        final Set<EntityTitleResponse> titles = new HashSet<>();
+        final Set<String> notPermitted = new HashSet<>();
+        documents.forEach(doc ->
+                {
+                    final String idAsString = doc.getObjectId("_id").toString();
+                    final boolean canReadTitle = checkCanReadTitle(subject, dbEntityCatalogEntry.get().readPermission(), idAsString);
+                    titles.add(
+                            new EntityTitleResponse(
+                                    idAsString,
+                                    collection,
+                                    canReadTitle ? doc.getString(titleField) : TITLE_IF_NOT_PERMITTED
+                            )
+                    );
+                    if (!canReadTitle) {
+                        notPermitted.add(idAsString);
+                    }
+                }
+        );
+
+        return new EntitiesTitleResponse(titles, notPermitted);
+    }
+
+    private boolean checkCanReadTitle(final Subject subject,
+                                      final String readPermission,
+                                      final String idAsString) {
+        return DbEntity.NO_PERMISSION.equals(readPermission) ||
```

Comment:
This makes all `DbEntity`s without an explicit permission configuration readable for all users, right?
I think it should be the other way around, not permitted by default.

What do you think?

Reply:
@luk-kaminski Hmm, good question. :thinking: How about a property on the annotation? (e.g., `@DbEntity(requiresReadPermission = false)` and default is `true`)

Reply:
@bernd - this would require handling a situation when someone states that permission is required, but does not provide this permission... or permission is not required, but provides this permission. 

I am seriously considering another constant (FORBIDDEN_ACCESS = "you:shall:not:pass!" ? ;) ), used as default.